core/src/sumeragi/fault.rs

```diff
@@ -526,7 +526,10 @@ impl<G: GenesisNetworkTrait, F: FaultInjection> SumeragiWithFault<G, F> {
             "Forwarding tx to leader"
         );
         // Don't require leader to submit receipts and therefore create blocks if the tx is still waiting for more signatures.
-        if let Ok(true) = tx.check_signature_condition(&self.wsv) {
+        if matches!(
```

Comment:
I think following looks cleaner:
```
if let Ok(MustUse(true)) = tx.check_signature_condition(&self.wsv) {
    ...
}
```

Reply:
The same problem with private field. I'm not sure why is it bad to make it public, but in the same time not sure why is it good. Looks like better matching is the only benefit

Reply:
Ok, decided to [implement it](https://github.com/hyperledger/iroha/pull/2582#discussion_r940535189)